qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -72,8 +73,8 @@ def _define(self):
 
         q = QuantumRegister(self.num_qubits, 'q')
         initialize_circuit = QuantumCircuit(q, name='init_def')
-        # TODO: make initialize an Instruction, and insert reset
```

Comment:
Can you derive Initialize from `Instruction`? Since it is not a Gate anymore (gates assumed to be reversible).
Also it should probably be called just `Initialize` now, rather than `InitializeGate`.

Reply:
If you dont mind, I would prefer to do that refactor in a different PR, called "Initialize is an instruction, not a gate". I prefer small PRs and the refactor is not directly related with #1896.

Reply:
It's just two lines I think. But this PR is inserting a reset in initialize, so it is converting it from a gate to an instruction, so it seems right that it becomes an Instruction in this PR.

Reply:
I tried... no 2 lines, but doable with little changes :) Done in 3ada9dd47b459bb092b485dc7fd751415ed04858